plugins/flytekit-vscode/flytekitplugins/vscode/decorator.py

```diff
@@ -0,0 +1,178 @@
+import multiprocessing
+import os
+import shutil
+import subprocess
+import sys
+import tarfile
+import time
+from functools import wraps
+from typing import Callable, Optional
+
+import fsspec
+
+from flytekit.loggers import logger
+
+# Where the code-server tar and plugins are downloaded to
+EXECUTABLE_NAME = "code-server"
+DOWNLOAD_DIR = "/tmp/code-server"
+HOURS_TO_SECONDS = 60 * 60
+DEFAULT_UP_SECONDS = 10 * HOURS_TO_SECONDS  # 10 hours
+
+
+def execute_command(cmd):
+    """
+    Execute a command in the shell.
+    """
+    process = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    logger.info(f"cmd: {cmd}")
+    stdout, stderr = process.communicate()
+    if process.returncode != 0:
+        raise RuntimeError(f"Command {cmd} failed with error: {stderr}")
+    logger.info(f"stdout: {stdout}")
+    logger.info(f"stderr: {stderr}")
+
+
+def download_file(url, target_dir="."):
+    """
+    Download a file from a given URL using fsspec.
+
+    Parameters:
+    - url (str): The URL of the file to download.
+    - target_dir (str, optional): The directory where the file should be saved. Defaults to current directory.
+
+    Returns:
+    - str: The path to the downloaded file.
```

Comment:
Minor nit pick:
Google style docstrings (which this doc string appears to be the closest to) would ommit the `-` in front of the parameters and return value.

Reply:
Good catch. Thank you!